pkg/agents/AgentTypes/agent2/agent2.go

```diff
@@ -0,0 +1,24 @@
+package agent2
+
+import (
+	"fmt"
+	baseagent "somas_base_platform/pkg/agents/BaseAgent"
+)
+
+type Agent2 struct {
+	*baseagent.BaseAgent
+}
+
+func (a2 *Agent2) Activity() {
+	fmt.Println("Agent2's Activity")
+	a2.BaseAgent.Activity()
+}
+
+func GetAgent() *Agent2 {
+	ag := &Agent2{
+		baseagent.GetAgent(),
```

Comment:
might be cleaner to have `GetAgent()` accept a name, i.e., `GetAgent("A2")`, which will allow you to initialise the name without a separate variable